processing/algs/BatchExportProjectComposers.py

```diff
@@ -0,0 +1,132 @@
+            )
+        )
+        self.addParameter(
+            QgsProcessingParameterEnum(
+                self.EXTENSION,
+                QCoreApplication.translate("BatchExportProjectComposers", "Extension for exported maps"),
+                ['.pdf', '.jpg', '.jpeg', '.tif', '.tiff', '.png', '.bmp', '.ico', '.ppm', '.xbm', '.xpm']
+            )
+        )
+        self.addParameter(
+            QgsProcessingParameterFolderDestination(
+                self.OUTPUT_FOLDER,
+                QCoreApplication.translate("BatchExportProjectComposers", "Output folder where to save maps")
+            )
+        )
+        self.addOutput(
+            QgsProcessingOutputFolder(
+                self.OUTPUT,
+                QCoreApplication.translate("BatchExportProjectComposers", 'Output')
+            )
+        )
+
+    def name(self):
+        return 'BatchExportProjectComposers'
+
+    def displayName(self):
+        return QCoreApplication.translate("BatchExportProjectComposers", "Batch export project composers")
+
+    def flags(self):
+        """ Important: this algorithm should run in the main thread """
+        return super().flags() | QgsProcessingAlgorithm.FlagNoThreading
+
+    def processAlgorithm(self, parameters, context, feedback):
+        Projects_folder = self.parameterAsString(parameters, self.PROJECTS_FOLDER, context)
+        Extension = self.parameterAsEnum(parameters, self.EXTENSION, context)
+        Output_folder = self.parameterAsString(parameters, self.OUTPUT_FOLDER, context)
+
+        projectPaths = glob.glob(os.path.join(Projects_folder, '*.qg[s|z]'))
+
+        count = 0
+        exported_count = 0
+
+        if not len(projectPaths):
+            feedback.reportError("\nERROR: No QGIS project files (.qgs or .qgz) found in the specified folder. We cannot continue...\n")
+        elif not os.path.isdir(Output_folder):
+            feedback.reportError("\nERROR: No output folder given. We cannot continue...\n")
+        else:
+            formats = ['.pdf', '.jpg', '.jpeg', '.tif', '.tiff', '.png', '.bmp', '.ico', '.ppm', '.xbm', '.xpm']
+
+            if not 'MapsPrinter' in qgis.utils.plugins:
+                feedback.reportError("\nERROR: The 'Maps Printer' plugin  is required!\n")
+            else:
+                mp = qgis.utils.plugins['MapsPrinter']
+                project = QgsProject.instance()
+                extension = formats[Extension]
+
+                # Do the work!
+                for projectPath in projectPaths:
+                    project.read(projectPath)
+                    feedback.pushInfo("\n'{}' project read!!".format(projectPath))
+                    feedback.setProgress(count * 100 / len(projectPaths))
+                    count += 1
+
+                    for composer in project.layoutManager().layouts():
+                        feedback.pushInfo("\n --> Composer found: '{}'!".format(composer.name()))
```

Comment:
while communicating with the user, can we use "layout" (I plan to offer report support one day) or "print layout" instead of "composer"?
```suggestion
                        feedback.pushInfo("\n --> Layout found: '{}'!".format(composer.name()))
```

Reply:
No problem here.
PR #19